src/bridge.ts

```diff
@@ -65,13 +66,21 @@ class DenoBridge {
     // a malformed semver range. If this does happen, let's throw an error so
     // that the tests catch it.
     if (downloadedVersion === undefined) {
-      throw new Error('Could not read downloaded binary')
+      const error = new Error(
+        'There was a problem setting up the Edge Functions environment. To try a manual installation, visit https://ntl.fyi/install-deno.',
+      )
+
+      if (this.onAfterDownload) {
+        this.onAfterDownload(error)
+      }
+
+      throw error
     }
 
     await this.writeVersionFile(downloadedVersion)
 
     if (this.onAfterDownload) {
-      this.onAfterDownload()
+      this.onAfterDownload(null)
```

Comment:
```suggestion
      this.onAfterDownload()
```

Together with the other suggestion.